### PR TITLE
Add HowTo on adding modules and plugins repositories

### PR DESCRIPTION
**Pull request by tdelhomenie-microej**

### CODE REVIEW

File: SDK6UserGuide/howtos/gradleAddRepository.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
+.. _sdk_6_howto_gradle_add_repository:
+
+How To Add A Repository
```
Comment (thomasponsmicroej) [inline SDK6UserGuide/howtos/gradleAddRepository.rst:3]:
```suggestion
How To Add a Repository
```
not 100% sure with case but seems consistent with the other titles.

File: SDK6UserGuide/howtos/gradleAddRepository.rst [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+The :ref:`SDK 6 installation process <sdk_6_configure_repositories>` asks to create a Gradle Init Script file to declare modules and plugins repositories.
+You may need to use additional repositories or replace the default ones, 
+for example to fetch a module only available in your company's repository.
+This page presents the different options to do that.
+
+If you need more details on this topic, 
+refer to `the official Gradle documentation on repository declaration <https://docs.gradle.org/current/userguide/declaring_repositories.html>`__.
+
+How to add a modules repository
```
Comment (thomasponsmicroej) [inline SDK6UserGuide/howtos/gradleAddRepository.rst:14]:
```suggestion
How To Add a Modules Repository
```

File: SDK6UserGuide/howtos/gradleAddRepository.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  The repositories defined here are fetched **after** the ones defined in the Gradle init script.
+
+- update the Gradle Init Script to add, replace or delete a repository.
+  The version of this script provided in the installation process is a recommended version to be applied to quickly setup an environment.
+  However, it can be modified to adapt it to your need, especially for the list of repositories.
+  The modules repositories are defined in the block ``settingsEvaluated > allprojects > repositories``, 
+  and are applied to all the Gradle builds executed on the machine.
+
+
+How to add a plugins repository
```
Comment (thomasponsmicroej) [inline SDK6UserGuide/howtos/gradleAddRepository.rst:50]:
```suggestion
How To Add a Plugins Repository
```

File: SDK6UserGuide/howtos/gradleAddRepository.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+refer to `the official Gradle documentation on repository declaration <https://docs.gradle.org/current/userguide/declaring_repositories.html>`__.
+
+How to add a modules repository
+-------------------------------
+
+The different ways to add a modules repository are:
+
+- for a simple project, add a ``repositories`` block in the ``build.gradle.kts`` of the project::
+
+    repositories {
```
Comment (thomasponsmicroej) [inline SDK6UserGuide/howtos/gradleAddRepository.rst:21]: could use a code block with kotlin syntax highligthing here?

File: SDK6UserGuide/howtos/gradleAddRepository.rst [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+	    maven {
```
Comment (thomasponsmicroej) [inline SDK6UserGuide/howtos/gradleAddRepository.rst:22]: indentation issue?

File: SDK6UserGuide/howtos/gradleAddRepository.rst [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        name = "myModulesRepository"
+        url = uri("https://my.company/my-modules-repository")
+      }
+    }
+  
+  The repositories defined here are fetched **after** the ones defined in the Gradle init script.
+
+- for a multi-project, add a ``subproject > repositories`` block in the ``build.gradle.kts`` of the multi-project::
```
Comment (thomasponsmicroej) [inline SDK6UserGuide/howtos/gradleAddRepository.rst:30]: if i don't put the subprojects block it won't apply to all projects of the multi project?

File: SDK6UserGuide/howtos/gradleAddRepository.rst [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  The version of this script provided in the installation process is a recommended version to be applied to quickly setup an environment.
+  However, it can be modified to adapt it to your need, especially for the list of repositories.
+  The modules repositories are defined in the block ``settingsEvaluated > allprojects > repositories``, 
+  and are applied to all the Gradle builds executed on the machine.
+
+
+How to add a plugins repository
+-------------------------------
+
+The different ways to add a modules repository are:
```
Comment (thomasponsmicroej) [inline SDK6UserGuide/howtos/gradleAddRepository.rst:53]:
```suggestion
The different ways to add a plugins repository are:
```

File: SDK6UserGuide/howtos/gradleAddRepository.rst [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+        }
+	    }
+    }
+
+  The repositories defined here are fetched **before** the ones defines in the init script.
+
+- update the Gradle Init Script to add, replace or delete a repository.
+  The version of this script provided in the installation process is a recommended version to be applied to quickly setup an environment.
+  However, it can be modified to adapt it to your need, especially for the list of repositories.
+  The modules repositories are defined in the block `settingsEvaluated > allprojects > repositories`,
```
Comment (thomasponsmicroej) [inline SDK6UserGuide/howtos/gradleAddRepository.rst:71]:
```suggestion
  The plugins repositories are defined in the block `settingsEvaluated > allprojects > pluginManagement > repositories`, 
```

File: SDK6UserGuide/howtos/gradleAddRepository.rst (hunk 1 above)
Comment (tdelhomenie-microej) [reply SDK6UserGuide/howtos/gradleAddRepository.rst]: Done.

File: SDK6UserGuide/howtos/gradleAddRepository.rst (hunk 2 above)
Comment (tdelhomenie-microej) [reply SDK6UserGuide/howtos/gradleAddRepository.rst]: Done.

File: SDK6UserGuide/howtos/gradleAddRepository.rst (hunk 3 above)
Comment (tdelhomenie-microej) [reply SDK6UserGuide/howtos/gradleAddRepository.rst]: Done.

File: SDK6UserGuide/howtos/gradleAddRepository.rst (hunk 4 above)
Comment (tdelhomenie-microej) [reply SDK6UserGuide/howtos/gradleAddRepository.rst:23]: Done, but it does not change anything.

File: SDK6UserGuide/howtos/gradleAddRepository.rst (hunk 5 above)
Comment (tdelhomenie-microej) [reply SDK6UserGuide/howtos/gradleAddRepository.rst]: Fixed.

File: SDK6UserGuide/howtos/gradleAddRepository.rst (hunk 6 above)
Comment (tdelhomenie-microej) [reply SDK6UserGuide/howtos/gradleAddRepository.rst]: Right, it is useless. So I removed the distinction between simple and multi projects, and I added a sentence about multi projects.

File: SDK6UserGuide/howtos/gradleAddRepository.rst (hunk 7 above)
Comment (tdelhomenie-microej) [reply SDK6UserGuide/howtos/gradleAddRepository.rst]: Done.

File: SDK6UserGuide/howtos/gradleAddRepository.rst (hunk 8 above)
Comment (tdelhomenie-microej) [reply SDK6UserGuide/howtos/gradleAddRepository.rst]: Done.

File: SDK6UserGuide/howtos/gradleAddRepository.rst [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+        }
+      }
+    }
+
+  The repositories defined here are fetched **before** the ones defines in the init script.
+
+- update the Gradle Init Script to add, replace or delete a repository.
+  The version of this script provided in the installation process is a recommended version to be applied to quickly setup an environment.
+  However, it can be modified to adapt it to your need, especially for the list of repositories.
+  The plugins repositories are defined in the block `settingsEvaluated > allprojects > pluginManagement > repositories`,
```
Comment (thomasponsmicroej) [inline SDK6UserGuide/howtos/gradleAddRepository.rst:65]: just missing a doubled ` around the expression I guess

**Review by thomasponsmicroej — APPROVED** [review]

File: SDK6UserGuide/howtos/gradleAddRepository.rst (hunk 9 above)
Comment (tdelhomenie-microej) [reply SDK6UserGuide/howtos/gradleAddRepository.rst]: Right, good catch. Done.

**Review by thomasponsmicroej — APPROVED** [review]

**Pull request merged**